---

**Instruções para o Projeto**

* Você deve importar o pandas e importar a base de dados da Polícia Militar do Rio de Janeiro sobre a violência
  * O arquivo `csv` se encontra no endereço abaixo:
    * https://raw.githubusercontent.com/carlosfab/curso_data_science_na_pratica/master/modulo_02/violencia_rio.csv
  * Este arquivo `csv` não está separado por vírgula, mas por ponto e vírgula. Por isso, deve ser importado da seguinte maneira:
    * `df = pd.read_csv("ENDEREÇO_DO_ARQUIVO_CSV", sep=";")`
  * Este é o site oficial, do Governo do Estado do Rio de Janeiro:
    * http://www.ispdados.rj.gov.br/estatistica.html
  * Um *dashboard* interativo e visual pode ser encontrado no site abaixo:
    * http://www.ispvisualizacao.rj.gov.br/index.html
  * Você está livre para apagar, alterar e acrescentar o que quiser!
  * Sem um guia, você vai sentir mais dificuldade, mas eu te garanto que se sentir fora da zona de conforto vai ser o mais importante no caminho do Data Science
  * Se você sentiu dificuldade, não se preocupe! Sexta-feira será disponibilizada a minha análise para você replicar exatamente como eu fiz.
    * Se sentir perdido é normal, mas acredite: O seu cérebro vai começar a adquirir uma capacidade nova de pensar em hipóteses, e questionar dados.
  
  
---
  

<center><img src="http://sigmoidal.ai/wp-content/uploads/2019/08/logo_color.png" height="40px"></center>

# Analisando a Violência no Rio de Janeiro

A violência no Brasil é um problema persistente que atinge direta ou indiretamente a população. O país tem níveis acima da média mundial no que se refere a crimes violentos, com níveis particularmente altos no tocante a violência armada e homicídios. Em 2016, o Brasil alcançou a marca histórica de 62 517 homicídios, segundo informações do Ministério da Saúde. Isso equivale a uma taxa de 30,3 mortes para cada 100 mil habitantes, uma das mais altas taxas de homicídios intencionais do mundo.O limite considerado como suportável pela Organização Mundial da Saúde (OMS) é de 10 homicídios por 100 mil habitantes.

## Obtenção dos Dados

O Instituto de Segurança Pública (ISP), criado pela Lei nº 3.329, de 28 de Dezembro 1999, é uma autarquia vinculada diretamente à Vice-Governadoria do Estado. Com 20 anos de existência, o ISP conta com grande conhecimento acumulado no desenvolvimento de metodologias de análise de dados relativos à Segurança Pública. Sua missão é produzir informações e disseminar pesquisas e análises com vistas a influenciar e subsidiar a implementação de políticas públicas de segurança e assegurar a participação social na construção dessas políticas.

Ética e transparência da informação são parte dos valores do Instituto: além de divulgar os dados relativos à segurança pública, o ISP realiza análises, projetos e pesquisas regularmente. As pesquisas visam a subsidiar a gestão de segurança pública no estado e o debate público sobre o tema, com o tratamento científico de seus principais problemas.

O ISP também é o responsável pela Coordenadoria dos Conselhos Comunitários de Segurança, canal de participação social e de comunicação entre a comunidade e os representantes das forças de segurança estaduais em cada região, na busca comum pela redução da violência.

http://www.ispdados.rj.gov.br/estatistica.html

### Importando os dados

Importe os dados, usando `df = pd.read_csv("endereço_do_csv", sep=";")`

Os dados podem ser obtidos a partir do seguinte link:https://raw.githubusercontent.com/carlosfab/curso_data_science_na_pratica/master/modulo_02/violencia_rio.csv


In [1]:
# importando pacotes necessários
import pandas as pd

In [ ]:
# gerando o dataframe
#df = pd.read_csv("https://raw.githubusercontent.com/carlosfab/curso_data_science_na_pratica/master/modulo_02/violencia_rio.csv", sep=";" , index_col=['vano','mes'])

In [7]:
# dados tratados com os campos nulos
df = pd.read_csv("https://raw.githubusercontent.com/juliosaopedro/Meus_Projetos/master/violencia_rio.csv", sep=";" , index_col=['vano','mes'])

## Análise Inicial dos Dados

Descreva e execute as seguintes etapas:

* Qual o tamanho do seu DataFrame (`shape`)
* Extrair e imprimir os nomes das colunas (`columns`)
  * Definição de algumas colunas
    * aaapai - Apreensão de Adolescente por Prática de Ato Infracional para jovens entre 12 e 17 anos    
    * apf - Auto de Prisão em Flagrante
    * cmp
    * cmba
  
* Quais os tipos das variáveis (`dtypes`)
* Visualizar as 5 primeiras linhas (`head`)
* Identifique a porcentagem de valores ausentes das colunas


In [ ]:
df.shape

In [ ]:
# exibindo as colunas
df.columns

In [ ]:
# tipos das colunas
df.dtypes

In [ ]:
# exibindo as 5 primeiras linhas
df.head(5)

In [8]:
# porcentagem de valores nulos
(df.isnull().sum() / df.shape[0] * 100).sort_values(ascending=False)

roubo_bicicleta               80.232558
furto_bicicleta               80.232558
apreensao_drogas_sem_autor    52.325581
posse_drogas                  52.325581
apf                           52.325581
aaapai                        52.325581
cmp                           52.325581
trafico_drogas                52.325581
cmba                          52.325581
estelionato                   41.860465
furto_celular                 41.860465
sequestro_relampago           41.860465
furto_coletivo                41.860465
roubo_apos_saque              41.860465
roubo_conducao_saque          41.860465
roubo_cx_eletronico           41.860465
extorsao                      41.860465
estupro                       41.860465
pol_militares_mortos_serv     41.860465
pol_civis_mortos_serv         41.860465
pessoas_desaparecidas         38.372093
encontro_ossada               38.372093
roubo_celular                 31.395349
ameaca                        27.906977
lesao_corp_morte              27.906977


## Informações Estatísticas da Violência no Rio de Janeiro

As informações aqui apresentadas cobrem dados do período de Janeiro de 1991 a Agosto de 2019.

Descreva e execute as seguintes etapas:

* Imprima o resumo estatístico do seu DataFrame (`describe`)
* Encontre as médias das seguintes variáveis (colunas):
  * `roubo_veiculo`
  * `furto_veiculos`
  * `recuperacao_veiculos`
* Calcule qual a porcentagem de carros recuperados em relação aos carros roubados + carros furtados:
  * $\frac{\text{roubo_veiculo} + \text{furto_veiculos}}{ \text{recuperacao_veiculos}}$
* Encontre os valores máximos (`max`) e mínimos (`min`) da coluna `hom_doloso`


In [ ]:
# resumo estatistico do dataframe
df.describe()

## Veículos
Roubos, Furtos e Recuperação

In [ ]:
# media de roubo de veiculos
df.roubo_veiculo.mean()

In [ ]:
# media de furto de veiculos
df.furto_veiculos.mean()

In [ ]:
# media da recuperaçao de veiculos
df.recuperacao_veiculos.mean()

In [ ]:
# porcentagem de veiculos recuperados em relacao aos roubos e furtos
df.recuperacao_veiculos.sum() / (df.roubo_veiculo.sum() + df.furto_veiculos.sum())  * 100

## Homicidio Doloso
O homicídio doloso é quando uma pessoa mata outra intencionalmente

In [ ]:
# valor minimo de homicidio doloso
df.hom_doloso.min()

In [ ]:
df.hom_doloso.mean()

In [ ]:
# valor maximo de homicidio doloso
df.hom_doloso.max()

## Visualização de Dados

Todos os dados aqui apresentados estão de acordo com a Lei de Acesso à Informação, objetivando maior transparência no que diz respeito às estatísticas de criminalidade e atividade policial do estado.(Instituto de Segurança Pública do Rio de Janeiro).

## Histograma

In [ ]:
# histograma de Homicidio Doloso
df['hom_doloso'].hist();

## Gráfico 

In [ ]:
# grafico de linhas para Homicidio Doloso
df.hom_doloso.plot();

## Roubo em Coletivo

In [ ]:
df.roubo_em_coletivo.plot();

## Conclusão

Alguns insights :

O percentual de recuperação de veiculos roubados ou furtados foi de 42% no período de Jan/1991 a Ago/2019.

O número de homicídios dolosos, apesar de ainda ser alto, vem diminuindo.

Já o roubo a coletivos vem aumentando

Os dados contém cerca de 20% de informação de roubo e furto de bicicleta